joomla/joomla-cms
# [4.0] Updated files and folders deletion on updates and remove 3.9.x update SQL scripts

Pull Request for Issue # .

### Summary of Changes

Updates the list of files and folders to be deleted to the changes since 4.0 Beta 5 and last changes in staging and 3.10-dev.

Add also files and folders having been forgotten with the release of 4.0 beta 2.

Changes comments for sections so they tell start and end point ("from" and "to") of the comparison.

Moved some Joomla 3.10 deletions form later sections to the first section and removed some duplicates.

Old J3 sample data images will not be deleted on update because they might be used in content, see discussion here https://github.com/joomla/joomla-cms/discussions/32744 .

The lists have been pre-generated with use of #25559 .

Order of folders within a release section changed to descending so it fits to what the tool produces because this makes sure the folders are delete from child to parent.

The `/libraries/vendor/paragonie/sodium_compat/src/Core/Xsalsa20.php` file is removed from this list of files to be deleted. This file has been renamed once from `Xsalsa20.php` to `XSalsa20.php` on Joomla 3 and is still present on Joomla 4 with the correct name. Having it in the list with the old name will cause it to be deleted on OS and files systems with case-insensitive file names. When using the script modified by PR #25559 , that file correctly doesn't appear on the list of files to be deleted.

In addition, this PR deletes from sources those new 3.9.x SQL update scripts which meanwhile have been merged up to 4.0-dev but should not be there.

### Testing Instructions

1. Code review by an experienced contributor or maintainer who knows the recent developments in J4:
Expand the changes in file `script.php` on GitHub and check that what has been added fits to the recent changes, like change of the build process for javascript or the change from Bootstrap 5.
=> Ping @dgrammatiko and @wilsonge .
2. Update test 3.10: Update from 3.10-dev or latest 3.10 nightly to the update package or custom update URL made by drone for this PR and compare the result on the file system with a new installation of current 4.0-dev or latest 4.0 nightly.
3. Verify that after the update from test 3 no file has been deleted which has been added to staging since 3.9.25 was released but which haven't been merged up to 4.0-dev yet.
4. Update tests 4.0 Beta: Update a 4.0 Beta 7 or an earlier 4.0 Beta to the update package or custom update URL made by drone for this PR and compare the result on the file system with a new installation of current 4.0-dev or latest 4.0 nightly.

### Actual result BEFORE applying this Pull Request

After an update from 3.10 or a previous 4.0 Beta to current 4.0-dev or latest 4.0 nightly, there are lots of obsolete files and folder left compared to a new installation of the same current 4.0-dev or latest 4.0 nightly, lots more than the exceptions mentioned below in section "Expected result AFTER applying this Pull Request".

E.g. this gazillion of `.es6.` files and their minified and gzipped versions being obsolete after the change of the build process, or all that old BS 4 stuff which should not be there.

On the other hand there are 2 "3.9.x" update SQL scripts for each database type after an update from 3.10, which should be deleted but they aren't.

### Expected result AFTER applying this Pull Request

After an update from 3.10 or a previous 4.0 Beta to the update package or custom update URL made by drone for this PR, there are only following old files and folders left compared to a new installation of current 4.0-dev.
1. Following folders including all sub folders and files are not deleted:
- /administrator/components/com_search
- /components/com_search
- /installation
- /modules/mod_search
- /plugins/fields/repeatable
- /plugins/search
2. Following files in other folders are not deleted:
- /administrator/language/en-GB/en-GB.com_search.ini
- /administrator/language/en-GB/en-GB.com_search.sys.ini
- /administrator/language/en-GB/en-GB.plg_fields_repeatable.ini
- /administrator/language/en-GB/en-GB.plg_fields_repeatable.sys.ini
- /administrator/language/en-GB/en-GB.plg_search_categories.ini
- /administrator/language/en-GB/en-GB.plg_search_categories.sys.ini
- /administrator/language/en-GB/en-GB.plg_search_contacts.ini
- /administrator/language/en-GB/en-GB.plg_search_contacts.sys.ini
- /administrator/language/en-GB/en-GB.plg_search_content.ini
- /administrator/language/en-GB/en-GB.plg_search_content.sys.ini
- /administrator/language/en-GB/en-GB.plg_search_newsfeeds.ini
- /administrator/language/en-GB/en-GB.plg_search_newsfeeds.sys.ini
- /administrator/language/en-GB/en-GB.plg_search_tags.ini
- /administrator/language/en-GB/en-GB.plg_search_tags.sys.ini
- /administrator/language/en-GB/en-GB.plg_search_weblinks.ini
- /administrator/language/en-GB/en-GB.plg_search_weblinks.sys.ini
- /administrator/language/en-GB/en-GB.plg_editors-xtd_weblink.ini
- /administrator/language/en-GB/en-GB.plg_editors-xtd_weblink.sys.ini
- /administrator/language/en-GB/en-GB.plg_system_weblinks.ini
- /administrator/language/en-GB/en-GB.plg_system_weblinks.sys.ini
- /language/en-GB/en-GB.com_search.ini
- /language/en-GB/en-GB.mod_search.ini
- /language/en-GB/en-GB.mod_search.sys.ini

There might be other files present after an update from 3.10 which are not mentioned in the list above. In this case there are files which have been recently added to the staging branch and merged up into 3.10-dev but not have been merged up into 4.10-dev yet.

There are no "3.9.x" update SQL scripts anymore after an update from 3.10.

### Documentation Changes Required

None.
